runtime/syntax/nftables.yaml

```diff
@@ -1,25 +1,23 @@
 filetype: nftables
 
 detect:
-    filename: "((nftables.conf$)|(nftables.rules.d/)|(nftables.d/)|(nftables.rules$))"
+   filename: "(nftables\\.(conf|rules)$|nftables(\\.rules)?\\.d/)"
     header: "^(#!.*/(env +)?nft( |$)|flush +ruleset)"
 
 rules:
     - type: "\\b(chain|counter|map|rule|ruleset|set|table)\\b"
     - type: "\\b(ether|inet|i(cm)?p(x|(v?(4|6))?)|tcp|udp)\\b"
     - special: "\\b(elements|hook|policy|priority|type|state)\\b"
-    - identifier: "\\b(ct|iif|iifname|meta|oif|oifname|th|dport|sport|saddr|daddr)\\b"
+    - identifier: "\\b(ct|iif|iifname|meta|oif|oifname|th|dport|sport|saddr|daddr|l4proto)\\b"
     - statement: "\\b(accept|drop|goto|jump|log|masquerade|reject|limit|queue)\\b"
     - preproc: "\\b(add|define|flush|include|delete)\\b"
-    - symbol.operator: "[<>.&|^!=:;,@]|[!=]|\\b(and|ge|gt|le|lt|or|xor)\\b"
+    - symbol.operator: "[<>.&|^:;,@]|[!=]|\\b(and|ge|gt|le|lt|or|xor)\\b"
     - constant.string: '([\+\-\*/][\"]{1})(.*)([\"]{1})'
     - constant.string:
          start: "\""
          end: "\""
       # Integer Constants
-    - constant.number: "(\\b([1-9][0-9]*|0[0-7]*|0[Xx][0-9A-Fa-f]+|0[Bb][01]+)([Uu][Ll]?[Ll]?|[Ll][Ll]?[Uu]?)?\\b)"
-      # Hexadecimal Floating Constants 
-    - constant.number: "(\\b0[Xx]([0-9A-Za-z]*[.][0-9A-Za-z]+|[0-9A-Za-z]+[.][0-9A-Za-z]*)[Pp][+-]?[0-9]+[FfLl]?\\b)"
```

Comment:
Here you removed hex at all. Why? Because hex is allowed in nftable rules.

Reply:
at which point will hex be allowed to use?

Reply:
They are allowed as simple integer values and checks against them.